src/main/java/CheckSum.kt

```diff
@@ -0,0 +1,101 @@
+enum class Encoding (val code: Int){
+    BECH32(1),
+    BECH32M(2)
+}
+
+
+
+object CheckSum {
+
+    private const val BECH32M_CONST = 0x2bc830a3
+
+    fun checkChecksum(humanPart: String, values: ByteArray): Boolean {
+        val humanPartExpanded = expandHumanPart(humanPart)
+        val combined = ByteArray(humanPartExpanded.size + values.size)
+        humanPartExpanded.copyInto(combined)
+        values.copyInto(combined, destinationOffset = humanPartExpanded.size)
+        return findPolynomial(combined) == 1
+    }
+
+    fun checkChecksum_bech32m(humanPart: String, values: ByteArray): Boolean {
```

Comment:
mixing camelcase and snake case, I would recommend choosing one and stick to it.